src/main/java/com/example/onlinebookstore/exceptions/CustomGlobalExceptionHandler.java

```diff
@@ -25,13 +38,13 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
         List<String> errors = ex.getBindingResult().getAllErrors().stream()
                 .map(this::getErrorMessage)
                 .toList();
-        BookErrorDto bookErrorDto = new BookErrorDto(
+        ErrorDto errorDto = new ErrorDto(
                 LocalDateTime.now(), errors);
-        return createResponseEntity(bookErrorDto);
+        return createResponseEntity(errorDto);
     }
 
-    private ResponseEntity<Object> createResponseEntity(BookErrorDto bookErrorDto) {
-        return new ResponseEntity<>(bookErrorDto, HttpStatus.BAD_REQUEST);
+    private ResponseEntity<Object> createResponseEntity(ErrorDto errorDto) {
+        return new ResponseEntity<>(errorDto, HttpStatus.BAD_REQUEST);
```

Comment:
you're setting `BAD_REQUEST` for each individual handler

this is a bad design - each handler might have specific status